framework/client/form.rst

```diff
@@ -0,0 +1,152 @@
+  provides the field with the requested props and gets the data from formField through the parent formField
+  FormDataContainer (point 7).
+
+  It returns the component ``<EditableValue>`` with these props:
+
+  * type
+  * value
+  * info
+  * events
+  * options
+
+2. formattedValueFactory
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  Analog to editableValueFactory for formatted values with only two props type and value.
+
+3. defaultMapping
+~~~~~~~~~~~~~~~~~~
+
+  Here the types of form fields get mapped and exported as an object consisting of type string
+  attributes and components as values. In this case values are often a call to editableValueFactory.
+  This way it’s possible to map the same call to editableValueFactory multiple times and provide the same
+  field to different types.
+
+  Example:
+
+  .. code-block:: console
+
+    ‘longitude’: editableValueFactory(‘coordinate’),
+    ‘latitude’: editableValueFactory(‘coordinate’)
+
+
+4. FormFieldFactory
+~~~~~~~~~~~~~~~~~~~~
+
+  FormFieldFactory receives the data for identifying the correct field from the redux-store through a redux container and
+  provides it to ``<ValueField/>``. The react-redux container is FormDataContainer (point 9). ``<ValueField/>`` returns the specific field
+  accessed through the mapping of the given type of the field.
+
+5. typeEditables
+~~~~~~~~~~~~~~~~~
+
+  The typeEditables directory contains files for different field types like ‘remote’. The files contain methods to overwrite
+  the default behaviour. This can be done through methods that get the events, value and options or data for
+  editableValueFactory. The data then is provided to each ``<EditableValue/>`` field.
+
+form
+-----
+
+6. FormBuilder
+~~~~~~~~~~~~~~~
+
+  The FormBuilder component traverses through the formDefinition data to resolve how to render each element. It  returns an
+  array of ``<Field/>`` components provided by the third party library redux-forms.
+  redux-forms requires to set the component prop to specify which component should be used to render the form field. To set
+  the ``<Field/>`` prop component the ``<ReduxFormFieldAdapter/>`` is used (point 7).
+
+7. ReduxFormFieldAdapter
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  The ``<ReduxFormFieldAdapter/>`` is the component that gets passed to ``<Field/>`` returns a call to formFieldFactory (point 2). redux-
+  forms allows to pass a factory as component prop. In addition to the props specific to the tocco-client other props from
+  redux-forms are passed as well. These props contain information about the state of the form e.g. if the form was edited or if
+  the user already clicked on a field.
+
+formData
+---------
+
+8. formData
+~~~~~~~~~~~~~
+
+  The formData parent directory contains the formData modules. Each module has its actions, reducers, and sagas to handle
+  the store data relating to data from the form.
+  formData then adds the data and methods to the store.
+
+  props:
+
+  - relationEntities
```

Comment:
a description for each "prop" would be nice.

Reply:
IMO the names are self-explanatory. I started with the descriptions and basically repeated what was already obvious by the names.